### PR TITLE
[autodiff] Support unary ops for forward mode

### PR DESCRIPTION
**Pull request by erizmr**
Related issue = #5055

### CODE REVIEW

File: taichi/transforms/auto_diff.cpp [hunk 1]
```diff
@@ -1115,10 +1115,32 @@ class MakeDual : public ADTransform {
       accumulate(stmt, mul(cos(stmt->operand), dual(stmt->operand)));
     } else if (stmt->op_type == UnaryOpType::cos) {
       accumulate(stmt, negate(mul(sin(stmt->operand), dual(stmt->operand))));
+    } else if (stmt->op_type == UnaryOpType::tan) {
+      TI_NOT_IMPLEMENTED
```
Comment (ailzhang) [inline taichi/transforms/auto_diff.cpp:1119]: OOC: why is this not implemented :P?  nit: semicolon at the end ;)

**Review by ailzhang — COMMENTED** [review]
Shall we add a few test cases in this PR as well?

File: taichi/transforms/auto_diff.cpp (hunk 1 above)
Comment (erizmr) [reply taichi/transforms/auto_diff.cpp:1119]: Sure. Adding test cases (and the semicolon ).

The reason for that is the derivative of `tan (x)` is `1/ cos^2(x)`, there are many singular points for this function which causing nan. lol

Comment (ailzhang) [reply taichi/transforms/auto_diff.cpp:1119]: aha would you mind adding this as a comment in the code? ;)

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**